include/glow/CodeGen/MemoryAllocator.h

```diff
@@ -119,35 +175,70 @@ class MemoryAllocator {
   /// Frees the allocation associated with \p handle.
   void deallocate(Handle handle);
 
-  /// \returns the high water mark for the allocated memory.
-  uint64_t getMaxMemoryUsage() const { return maxMemoryAllocated_; }
+  /// \returns the total size (in bytes) of the memory.
+  uint64_t getMemorySize() const { return memorySize_; }
 
-  /// \returns the size of the whole memory region that we can allocate segments
-  /// into.
-  uint64_t getMemorySize() const { return poolSize_; }
+  /// \returns the maximum memory usage (in bytes).
+  uint64_t getMaxMemoryUsage() const { return maxUsedSize_; }
 
   /// \returns the alignment boundary used to align segments.
   size_t getAlignment() const { return alignment_; }
 
+  /// \returns the allocation efficiency as a float between 0.0 and 1.0
+  /// which quantifies the efficiency of the allocation algorithm. An
+  /// efficiency value of 1.0 means the best theoretically possible. The
+  /// efficiency is not always 1.0 due to memory fragmentation.
+  float getAllocationEfficiency() const;
+
   /// \returns the name of the memory region.
   const std::string &getName() const { return name_; }
 
+  /// Function to virtually increase the memory usage of the allocator with
+  /// \p size (in bytes). \returns the new memory usage or MemoryAllocator::npos
+  /// if the new memory usage exceeds the memory size. The given \p size when
+  /// used internally is rounded up to nearest multiple of the alignment.
+  uint64_t addMemoryUsage(uint64_t size);
```

Comment:
@mciprian13 Quick question: You can increase memory usage this way, but it it not allocated to any object, right? And it is not possible to deallocate? So, it is very unusual in this sense. I think you should either clearly document this in the comment or even better make it possible to deallocate this memory somehow.

Another question: Do we need the opposite of this function so that we have incMemoryUsage(uint64_t)/decMemoryUsage(uint64_t)? if so, decMemoryUsage should probably check if any allocated segments are affected by this and then fail or free them.